packages/help-center/src/components/help-center.tsx

```diff
@@ -2,51 +2,136 @@
+			setThirdPartyCookiesAllowed( true );
+		} else if ( zendeskStatus === 'error' ) {
+			setThirdPartyCookiesAllowed( false );
+		}
+	}, [ setThirdPartyCookiesAllowed, zendeskStatus ] );
+
+	useEffect( () => {
+		if ( ! chatStatus?.is_user_eligible ) {
+			return;
+		}
+
+		const zendeskKey: string | false = config( 'zendesk_support_chat_key' );
+		if ( ! zendeskKey ) {
+			return;
+		}
+
+		if ( document.getElementById( ZENDESK_SCRIPT_ID ) ) {
+			return;
+		}
+
+		function setUpMessagingEventHandlers() {
+			if ( typeof window.zE !== 'function' ) {
+				return;
+			}
+
+			window.zE( 'messenger', 'hide' );
+
+			window.zE( 'messenger:on', 'open', function () {
+				setShowMessagingWidget( true );
+			} );
+			window.zE( 'messenger:on', 'close', function () {
+				setShowMessagingWidget( false );
+			} );
+			window.zE( 'messenger:on', 'unreadMessages', function ( count ) {
+				if ( Number( count ) > 0 ) {
+					setShowMessagingLauncher( true );
+				}
+			} );
+		}
 
-	const { unreadCount, closeChat } = useHCWindowCommunicator( isMinimized || ! show );
+		loadScript(
+			'https://static.zdassets.com/ekr/snippet.js?key=' + encodeURIComponent( zendeskKey ),
+			setUpMessagingEventHandlers,
+			{ id: ZENDESK_SCRIPT_ID }
+		);
+	}, [ setShowMessagingLauncher, setShowMessagingWidget, chatStatus ] );
 
+	const { data: messagingAuth } = useMessagingAuth( Boolean( chatStatus?.is_user_eligible ) );
 	useEffect( () => {
-		setUnreadCount( unreadCount );
-	}, [ unreadCount, setUnreadCount ] );
+		const jwt = messagingAuth?.user.jwt;
+		if ( typeof window.zE !== 'function' || ! jwt ) {
+			return;
+		}
+
+		window.zE( 'messenger', 'loginUser', function ( callback ) {
+			callback( jwt );
+		} );
+	}, [ messagingAuth ] );
+
+	const { showMessagingLauncher, showMessagingWidget } = useSelect( ( select ) => {
+		const helpCenterSelect: HelpCenterSelect = select( HELP_CENTER_STORE );
+		return {
+			showMessagingLauncher: helpCenterSelect.isMessagingLauncherShown(),
+			showMessagingWidget: helpCenterSelect.isMessagingWidgetShown(),
+		};
+	}, [] );
+
+	useEffect( () => {
+		if ( typeof window.zE !== 'function' ) {
+			return;
+		}
+		if ( showMessagingLauncher ) {
+			window.zE( 'messenger', 'show' );
+		} else {
+			window.zE( 'messenger', 'hide' );
+		}
+	}, [ showMessagingLauncher ] );
+
+	useEffect( () => {
+		if ( typeof window.zE !== 'function' ) {
+			return;
+		}
+		if ( showMessagingWidget ) {
+			window.zE( 'messenger', 'open' );
+		} else {
+			window.zE( 'messenger', 'close' );
+		}
+	}, [ showMessagingWidget ] );
 
+	const email = useSelector( getCurrentUserEmail );
+	const { data: supportHistory } = useSupportHistory( 'ticket', email );
 	useEffect( () => {
-		if ( data?.status === 'assigned' ) {
-			setShowHelpCenter( true );
+		if ( supportHistory?.some( ( ticket ) => ticket.channel === 'native_messaging' ) ) {
+			setShowMessagingLauncher( true );
 		}
-	}, [ data, setShowHelpCenter ] );
+	}, [ setShowMessagingLauncher, supportHistory ] );
```

Comment:
Same as the other effect, we can add a `select` in `useSupportHistory` that returns a boolean.

Reply:
I appreciate the suggestion, but in this case I prefer to keep it as-is. `useSupportHistory` returns an array of previous support interactions of given type - plain and simple. Throwing in an additional field there would make its return value less clear. And technically, it can also request a `chat` support history (though, with the advent of Zendesk Messaging for chat that needs cleaning up cause Messaging "chats" are actually tickets too), where this property would not apply.